src/app.ts

```diff
@@ -2,942 +2,990 @@ import {
-
-    addModule({
-      name: "hasStorageAccess",
-      initializedRequired: true,
-      hasOutput: true,
-      onClick:  async () => {
-        const perm = await document.hasStorageAccess();
-        console.log(perm);
-        return perm;
-      },
-    });
+const initializeAppModules = () => {
+  var childWindow;
+  let totalStates = 0;
+  let attachmentArray;
 
-    addModule({
-      name: "popOut",
-      initializedRequired: true,
-      hasOutput: true,
-      onClick:   () => {
-        window.open(window.location.href);
-      },
-    });
-    
-    addModule({
-      name: "setUnpartitionedCookie",
-      initializedRequired: true,
-      hasOutput: true,
-      onClick:  async () => {
-        document.cookie = document.cookie + `myCookie${Math.random()%100}=helloworld`;
-        return document.cookie;
-      },
-    });
+  initialieCacheToggleButton();
+  initializeDownloadLinks();
+  outputTabRenderedLocation(microsoftTeams.getContext);
 
-    addModule({
-      name: "readUnpartitionedCookie",
-      initializedRequired: true,
-      hasOutput: true,
-      onClick:  async () => {
-        console.log(document.cookie);
-        return document.cookie;
-      },
-    });
-    
-
-    addModule({
-      name: "requestStorageAccess",
-      initializedRequired: true,
-      hasOutput: true,
-      onClick:  async () => {
-        const perm = await document.requestStorageAccess();
-        console.log(perm);
-        return perm;
-      },
-    });
-
-    addModule({
-      name: "enablePrintCapability",
-      initializedRequired: true,
-      hasOutput: false,
-      action: function () {
-        microsoftTeams.enablePrintCapability();
-      },
-    });
+  addModule({
+    name: "hasStorageAccess",
+    initializedRequired: true,
+    hasOutput: true,
+    onClick:  async () => {
+      const perm = await document.hasStorageAccess();
+      console.log(perm);
+      return perm;
+    },
+  });
 
-    addModule({
-      name: "print",
-      initializedRequired: true,
-      hasOutput: false,
-      action: function () {
-        microsoftTeams.print();
-      },
-    });
+  addModule({
+    name: "popOut",
+    initializedRequired: true,
+    hasOutput: true,
+    onClick:   () => {
+      window.open(window.location.href);
+    },
+  });
+  
+  addModule({
+    name: "setUnpartitionedCookie",
+    initializedRequired: true,
+    hasOutput: true,
+    onClick:  async () => {
+      document.cookie = document.cookie + `myCookie${Math.random()%100}=helloworld`;
+      return document.cookie;
+    },
+  });
 
-    addModule({
-      name: "app.getContext",
-      initializedRequired: true,
-      hasOutput: true,
-      action: function (output) {
-        microsoftTeams.app.getContext().then(output);
-      },
-    })
-
-    addModule({
-      name: "getContext",
-      initializedRequired: true,
-      hasOutput: true,
-      action: function (output) {
-        microsoftTeams.getContext(output);

```

Comment:
It might be best to keep both the new and old version of APIs (at least this one) since they actually behave differently. The v1 version of getContext returns a flat list of props while the v2 version returns props organized in logical groupings.

Could we maybe have a toggle that lets you flip between v1 and v2? We do need to make sure that the v1 versions continue to function since many apps still use them.

Reply:
I've reverted the change, since there is already another test using app.getContext.